File: Python/Zadanie 2/szachy_python/pieces/knight.py
from piece import Piece

class Knight(Piece):
    def symbol(self):
        return '♘' if self.color == 'white' else '♞'

    def is_valid_move(self, board, start, end):
        dx = abs(end[0] - start[0])
        dy = abs(end[1] - start[1])
        if (dx, dy) not in [(2, 1), (1, 2)]:
            return False
        target = board.get_piece(end)
        return target is None or target.color != self.color

File: C#/Zadanie 2/Szachy/Knight.cs
namespace Szachy;

public class Knight : Piece
{
    public Knight(PieceColor color) : base(color) {}

    public override char Symbol => Color == PieceColor.White ? 'N' : 'n';

    public override bool IsValidMove((int x, int y) from, (int x, int y) to, Board board)
    {
        int dx = Math.Abs(to.x - from.x);
        int dy = Math.Abs(to.y - from.y);

        if ((dx == 2 && dy == 1) || (dx == 1 && dy == 2))
        {
            var target = board.Squares[to.x, to.y];
            return target == null || target.Color != this.Color;
        }

        return false;
    }
}